frontend/public/components/monitoring/alert-manager-receiver-forms.tsx

```diff
@@ -0,0 +1,500 @@
+} from './alert-manager-utils';
+import { K8sResourceKind } from '../../module/k8s';
+import { ActionGroup, Button } from '@patternfly/react-core';
+import { Helmet } from 'react-helmet';
+import { RadioInput } from '../radio';
+import { MinusCircleIcon, PlusCircleIcon } from '@patternfly/react-icons';
+import { AlertManagerConfig, AlertManagerReceiver } from './alert-manager-config';
+
+const PagerDutySubForm = ({ pagerDutyValues, setPagerDutyValues }) => {
+  const updatePagerDutyValues = (key, value) => {
+    _.set(pagerDutyValues, key, value);
+    setPagerDutyValues(_.clone(pagerDutyValues));
+  };
+
+  return (
+    <div className="co-m-pane__body--section-heading">
+      <SectionHeading text="PagerDuty Configuration" />
+      <div className="form-group">
+        <label className="control-label">Integration Type</label>
+        <div>
+          <RadioInput
+            title="Events API v2"
+            name="integrationType_Events"
+            id="integration-type-events"
+            value="events"
+            onChange={(e) => updatePagerDutyValues('integrationType', e.target.value)}
+            checked={pagerDutyValues.integrationType === 'events'}
+            inline
+          />
+          <RadioInput
+            title="Prometheus"
+            name="integrationType_Prometheus"
+            id="integration-type-prometheus"
+            value="prometheus"
+            onChange={(e) => updatePagerDutyValues('integrationType', e.target.value)}
+            checked={pagerDutyValues.integrationType === 'prometheus'}
+            inline
+          />
+        </div>
+      </div>
+      <div className="form-group">
+        <label className="control-label co-required">
+          {pagerDutyValues.integrationType === 'events' ? 'Routing' : 'Service'} Key
+        </label>
+        <input
+          className="pf-c-form-control"
+          type="text"
+          aria-describedby="integration-key-help"
+          name="integrationKey"
+          id="integration-key"
+          value={pagerDutyValues.integration_key}
+          onChange={(e) => updatePagerDutyValues('integrationKey', e.target.value)}
+        />
+        <div className="help-block" id="integration-key-help">
+          PagerDuty integration key
+        </div>
+      </div>
+    </div>
+  );
+};
+
+const WebhookSubForm = ({ webhookUrl, setWebhookUrl }) => {
+  return (
+    <div className="form-group">
+      <label className="control-label co-required">URL</label>
+      <input
+        className="pf-c-form-control"
+        type="text"
+        aria-describedby="webhook-url-help"
+        name="webhookUrl"
+        id="webhook-url"
+        value={webhookUrl}
+        onChange={(e) => setWebhookUrl(e.target.value)}
+      />
+      <div className="help-block" id="webhook-url-help">
+        The endpoint to send HTTP POST requests to
+      </div>
+    </div>
+  );
+};
+
+const RoutingLabels = ({ routeLabels, setRouteLabels, saveAsDefaultReceiver }) => {
```

Comment:
This looks almost identical to the Matchers section of the Create / Edit Silence page, so it would be better to use the same code for both.

Reply:
It is similar, but not really.  The one on the Silence page is an older style `class` and stores the labels and all other values in an object.  This one is using webhooks and shows/hides certain form fields, and sets certain defaults, when dealing with the default Receiver.

Reply:
Thanks for the explanation. Once the Silence page is updated to use webhooks, perhaps we can extract the common logic. There is also `NameValueEditor`, with overlapping functionality too.